examples/07_advanced/plot_beta_series.py

```diff
@@ -0,0 +1,175 @@
+"""
+Beta-Series Modeling for Task-Based Functional Connectivity and Decoding
+========================================================================
+This example shows how to run :term:`GLM`s for beta series models, which are a
+common modeling approach for a variety of analyses of task-based fMRI
+data with an event-related task design, including functional connectivity,
+decoding, and representational similarity analysis.
+
+Beta series models fit trial-wise conditions, which allow users to create
+"time series" of these trial-wise maps, which can be substituted for the
+typical time series used in resting-state functional connectivity analyses.
+Generally, these models are most useful for event-related task designs,
+while other modeling approaches (e.g., PPI) tend to perform better in block
+designs, depending on the type of analysis.
+See [Cisler, Bush, & Steele (2014)]
+(https://doi.org/10.1016/j.neuroimage.2013.09.018)
+for more information about this, in the context of functional connectivity
+analyses.
+
+Two of the most well-known beta series modeling methods are
+Least Squares- All (LSA) and Least Squares- Separate (LSS).
+In LSA, a single :term:`GLM` is run, in which each trial of each condition of
+interest is separated out into its own condition within the design matrix.
+In LSS, each trial of each condition of interest has its own :term:`GLM`,
+in which the targeted trial receives its own column within the design matrix,
+but everything else remains the ssame as the standard model.
+Trials are then looped across, and many GLMs are fitted,
+with the parameter estimate map extracted from each GLM to build the LSS beta
+series.
+"""
+from nilearn.glm.first_level import FirstLevelModel
+from nilearn import image, masking, plotting
+
+##############################################################################
+# Prepare data and analysis parameters
+# -------------------------------------
+# Prepare the data.
+from nilearn.datasets import fetch_language_localizer_demo_dataset
+
+data_dir, _ = fetch_language_localizer_demo_dataset()
+
+from nilearn.glm.first_level import first_level_from_bids
+
+task_label = 'languagelocalizer'
+models, models_run_imgs, events_dfs, models_confounds = \
+    first_level_from_bids(
+        data_dir,
+        task_label,
+        img_filters=[('desc', 'preproc')],
+    )
+
+standard_glm = models[0]
+fmri_file = models_run_imgs[0][0]
+events_df = events_dfs[0][0]
+confounds_df = models_confounds[0][0]
+
+mask = masking.compute_epi_mask(fmri_file)
+
+# We will use :func:`~nilearn.glm.first_level.first_level_from_bids`'s
+# parameters for the other models.
+glm_parameters = standard_glm.get_params()
+# We need to override one parameter (signal_scaling) with the value of
+# scaling_axis.
+glm_parameters["signal_scaling"] = standard_glm.scaling_axis
```

Comment:
The reason for that is not obvious  to me.

Reply:
The parameter `signal_scaling` must be an integer for the axis, but after initialization that integer is stored as `scaling_axis` and the saved parameter for `signal_scaling` is a boolean, so if you grab the saved parameter and feed it directly into a new FirstLevelModel, it will raise an exception when it tries to standardize based on a boolean.

Here's a way to reproduce it:

```python
from nilearn.datasets import fetch_language_localizer_demo_dataset
from nilearn.glm.first_level import FirstLevelModel, first_level_from_bids

data_dir, _ = fetch_language_localizer_demo_dataset()
models, models_run_imgs, events_dfs, models_confounds = \
    first_level_from_bids(
        data_dir,
        'languagelocalizer',
        img_filters=[('desc', 'preproc')],
    )
standard_glm = models[0]
fmri_file = models_run_imgs[0][0]
events_df = events_dfs[0][0]

# Grab the parameters
glm_parameters = standard_glm.get_params()

# We can fit the original GLM just fine
standard_glm.fit(fmri_file, events_df)

# But not a new one
new_glm = FirstLevelModel(**glm_parameters)
new_glm.fit(fmri_file, events_df)
```

Here's the traceback:

```Text
TypeError                                 Traceback (most recent call last)
<ipython-input-1-86ffa75b5c7d> in <module>
     21 # But not a new one
     22 new_glm = FirstLevelModel(**glm_parameters)
---> 23 new_glm.fit(fmri_file, events_df)

~/Documents/tsalo/nilearn/nilearn/glm/first_level/first_level.py in fit(self, run_imgs, events, confounds, design_matrices, bins)
    568 
    569             if self.signal_scaling:
--> 570                 Y, _ = mean_scaling(Y, self.scaling_axis)
    571             if self.memory:
    572                 mem_glm = self.memory.cache(run_glm, ignore=['n_jobs'])

~/Documents/tsalo/nilearn/nilearn/glm/first_level/first_level.py in mean_scaling(Y, axis)
     59 
     60     """
---> 61     mean = Y.mean(axis=axis)
     62     if (mean == 0).any():
     63         warn('Mean values of 0 observed.'

/opt/miniconda3/lib/python3.8/site-packages/numpy/core/_methods.py in _mean(a, axis, dtype, out, keepdims, where)
    177             is_float16_result = True
    178 
--> 179     ret = umr_sum(arr, axis, dtype, out, keepdims, where=where)
    180     if isinstance(ret, mu.ndarray):
    181         ret = um.true_divide(

TypeError: an integer is required
```

Reply:
Then this is a bug. We should open an issue for that.

Reply:
I'm going to resolve this thread since I opened #3134. I'll update this branch once #3135 is merged.